### PR TITLE
Survival Revival

### PR DESCRIPTION
**Pull request by Swagile**
## About The Pull Request

This is a revival of survivors with a purpose behind them. This brings risk and reward to both factions (marines and xenos). Survivors now start in a bunker deep in enemy territory (caves) with shutters that protect them for 10 minutes while being provided a decent amount of supplies to allow them to hold against incoming threats. The bunker comes with a fully equipped medbay (minus the autodoc and cryo tube), an armory, and a engineering area. There are also surplus supplies available, such as MRE's, water, flares, and engi materials. Once 10 minutes pass, the shutters go down forever, and survivors must defend against xenos if xenos choose to attack the bunker.

The reward for attacking and defeating the survivors is 2 free generators for psy on Distress (and Hunt if Bravemole adds back psy to that gamemode), and survivors giving 1 larvae per survivor that spawns in (on Hunt AND Distress).

Made this draft PR as im not done yet and want to see if maintainers want me to do anything specifically before I go forward (like splitting PR up into multiples for each map edit, etc).

Todo:

1) Remove all survivor spawns and consolidate them into bunkers.
2) Do bunkers on all maps (Barrenquila and LV are done so far).
3) Make a intel item that, when sent down ASRS pad, gives 200 points. Spawn 1 in bunker and make it acidable and non-storageable; have to hand carry it to FOB / Tadpole for evac back to ship.
4) Maybe make survivor gens special, with them giving more than other generators? Idk.
5) Make each 1 survivor that spawns in give 1 larvae to xenos regardless of gamemode (or give more psy in Distress? Idk what would be a better solution).

## Why It's Good For The Game

Gives survivors a "purpose" in the game, and gives xenos and marines a wild card back into the game. Mechanically stops survivors from rushing down xenos at start and xenos know exactly where they are every round, meaning they can be prepared for them. Gives combat xenos something to do during prep, and if survivors are good enough, gives marines a reward for actually saving them. Gives us back survivor if we only want to play for 30 mins or so, or just want a challenging experience. Balance them so that unless xenos are really bad, they won't get full wiped by survivors.

## Changelog
:cl:
add: Added new things
add: Added more things
del: Removed old things
qol: made something easier to use
balance: rebalanced something
fix: fixed a few things
soundadd: added a new sound thingy
sounddel: removed an old sound thingy
imageadd: added some icons and images
imagedel: deleted some icons and images
spellcheck: fixed a few typos
code: changed some code
refactor: refactored some code
config: changed some config setting
admin: messed with admin stuff
server: something server ops should know
/:cl:

### CODE REVIEW

**Comment by Swagile** [comment]
ill fix the linter tomorrow, 2 lazy 2 do it today

**Comment by InterroLouis** [comment]
Pictures of the bunkers would be nice.

**Comment by DerrickChamberlain** [comment]
Looking good so far my man

**Comment by David-Stormwell** [comment]
Remove survivor rambo armor, delete marine+ guns which spawn in maps such as lava outpost and the such.
Optionally make intel computers at round start activated which gives survivors an option to push out and grab them.

**Comment by Jeff-jefferson-jeffery** [comment]
if you're removing their rambo armour, make sure to give them some sort of suit slot item that can fit guns so they're not completely fucked, maybe bulletproof vests, because those are relatively weak stat wise

**Comment by David-Stormwell** [comment]
> 
> 
> if you're removing their rambo armor, make sure to give them some sort of suit slot item that can fit guns so they're not completely fucked, maybe bulletproof vests, because those are relatively weak stat-wise

This comes into another thing.
While editing maps, make sure sec and other lockers dotted around logically have armor.
Doesn't have to be good armor, just enough to be decent enough yet not as overpowered as rambo was.

**Comment by SolarMacharius** [comment]
This prevents the primary issue of survivors ramboing and killing all the xenos while making playing survivor still viable. This is a fantastic idea for a rework and gives xenos something to do before the marines arrive with actual risks and rewards. Great idea.

**Comment by psykzz** [comment]
> Pictures of the bunkers would be nice.

image
Full picture
https://mdb.affectedarc07.co.uk/Files/158907057/2580408610/1/after.png

**Comment by psykzz** [comment]
You could potentially make use of the procedural maps to move the bunker each round as well

File: code/datums/gamemodes/game_mode.dm [hunk 1]
```diff
@@ -303,6 +303,10 @@ Sensors indicate [numXenosShip || "no"] unknown lifeform signature[numXenosShip
 		addtimer(CALLBACK(GLOBAL_PROC, .proc/send_global_signal, COMSIG_GLOB_OPEN_TIMED_SHUTTERS_LATE), SSticker.round_start_time + 30 MINUTES)
 			//Called late because there used to be shutters opened earlier. To re-add them just copy the logic.
 
+	if(flags_round_type & MODE_LZ_SHUTTERS)
```
Comment (BraveMole) [inline code/datums/gamemodes/game_mode.dm:306]: add another flag for it. MODE_SURVIVOR or whatever, and give it to hunt and distress

**Review by BraveMole — CHANGES_REQUESTED** [review]

File: code/datums/gamemodes/game_mode.dm (hunk 1 above)
Comment (Swagile) [reply code/datums/gamemodes/game_mode.dm:306]: will do after i fix da linter

**Comment by Swagile** [comment]
> 
> 
> You could potentially make use of the procedural maps to move the bunker each round as well

i would but:

a) i have never done procedural maps so id have to figure that out and

b) there are so few places to put the bunker due to its size and 

c) it is that size so that it can reasonably hold all its facilities and be big enough so you can cade it decently without it being a cluster fuck and dying to one boiler blob

**Comment by Alterist** [comment]
```
 _maps/map_files/LV624/LV624.dmm:14266:/area/lv624/ground/survival{
Vareditted /area path use detected in maps, please replace with proper paths.
```

**Comment by Swagile** [comment]
> 
> 
> Remove survivor rambo armor, delete marine+ guns which spawn in maps such as lava outpost and the such.
> Optionally make intel computers at round start activated which gives survivors an option to push out and grab them.

No, and no.

Reason number 1: I am nerfing the chest part for rambo armor, but the rest is fine. Survivors are outnumbered, xenos know where they are with this PR, and xenos also have 10 minutes to prepare for them. If they can't beat the survivors with these conditions despite nerfed chest armor then they weren't going to win against marines anyway.

Reason number 2: No partially due to the above; xenos have 10 mins to prepare so they are fully evo'd and can even build around their bunker if they have spare castes to do so and wish to challenge survivors. If survivors, in these conditions, manage to somehow break out of survivor bunker area and risk themselves in caves to then make it to the colony portions of the map for better armor and weapons, then by all means, they should be rewarded for it. This also leaves the bunker open for invasion; while survivors leave to grab better armor / guns, xenos can just destroy and mass acid everything in survivor bunker, and claim the gens for free while leaving survivors nowhere to go except to roam.

**Comment by Tupinambis** [comment]
This is a very interesting and fun idea that seems to be well thought out. They probably shouldn't be too deep in enemy territory though. To suggest some spots:

LV-624: Sand Temple. Alternatively a secure dome northwest colony. 
Big Red: Northeast dormitories where the engineering and cargo rooms are. 
Magmoor: Research or Hydro. These buildings are never visited otherwise. 
Ice Colony: The engineering/cargo/science building in the southwest of the map, directly north of underground. 
Prison Station: Crashed Ship, kind of an obvious one.

**Comment by TiviPlus** [comment]
Concept looks ok to me

**Comment by InterroLouis** [comment]
Are those reinforced walls, or are there any heavy rwalls in there? It seems like it would be pretty easy for xenos to just ignore the front with the doors and melt in from the back every time if they're just regular R walls.

**Comment by SplinterGP** [comment]
survivors already give some surplus to benos when they spawn i believe

**Comment by SplinterGP** [comment]
1- how does the free generator work, what if there are no survs
2- i dont like survs impacting on rest of game unless they survive, even more since marines can't do anything about it as they can't go down until X time.
3- unless you give survs 30000000 ammo i doubt they'll ever defend that OR even more stay on that bunker, which staying on a single spot as a survivor is literally waiting to die as you run out of supplies, plus depending of gamemode if it has turrets just spam turrets and stuff around the bunker, fobcamp style

**Comment by Swagile** [comment]
> 
> 
> 1- how does the free generator work, what if there are no survs
> 2- i dont like survs impacting on rest of game unless they survive, even more since marines can't do anything about it as they can't go down until X time.
> 3- unless you give survs 30000000 ammo i doubt they'll ever defend that OR even more stay on that bunker, which staying on a single spot as a survivor is literally waiting to die as you run out of supplies, plus depending of gamemode if it has turrets just spam turrets and stuff around the bunker, fobcamp style

If there's no survs it works like normal; its a free gen. I dunno how id make it so they don't appear if there are no survivors.

Survivors being an impact on the game is a condition for their revival; one of the stated reasons for survivor removal is their only purpose is "spawn in, grab guns and meds as fast as possible, rush xenos". Now survs get a base but can't leave it for 10 mins and in return if they survive they benefit marines; if they don't survive, they benefit xenos (extra gens).

The survivor bunker has:
20 MP5 mags
3 MP5's already with mags
2 extended MP5 mags
4 AP MP5 mags
2 paladin shotguns
1 Double Barrel shotgun
1 Incendiary slug box
A dozen buck, flech, and slug boxes (not the big boxes, just small ones with the 5 handfuls)
1 flamethrower
1 mosin nagant with ammo packets

This doesn't include the other things I gave them, such as engi mats, and 2 mini sentries.

This is enough for a moderately skilled survivor team to pose a challenge to xenos if they use everything at their disposal. Of course, these can be adjusted if this is too much, but personally (from experience playing survivor and as a xeno against survivor) this is more than enough ammo and weaponry for survivors with experience to hold. Good survivors have held for 10-15 minutes with less than that (see: LV Secure Storage holds).

If xenos just spam turrets in front of the survivor bunker, survivors can just flamethrower them and then mosin nagant them from afar. This is a waste of xeno psy points against survivors with even 1 IQ with the supplies I have put in the survivor bunker.

**Comment by Swagile** [comment]
> 
> 
> Are those reinforced walls, or are there any heavy rwalls in there? It seems like it would be pretty easy for xenos to just ignore the front with the doors and melt in from the back every time if they're just regular R walls.

R-walls.

The way ive balanced the bunkers is that there is 1 main entrance (south) and one side entrance (so far east).

This means one entrance is wide open and survivors can easily defend it while the other is acidable by boiler or queen. Every other entrance has destructible rocks (the PC only type).

It makes the bunker not an absolute choke so xenos stand a chance, but survs can also survive since the supplies ive given are ample enough.

**Comment by Swagile** [comment]
Gunna figure out how to use signals to qdel the (future) intel req item and current gens if no survs spawn. Till then this is on hold.

**Comment by JackTheJackhammer** [comment]
Wait, whats the purpose/incentive for going survivor? Xenos have all the benefit from what I read. They can choose to attack. If the survivors leave the bunker, they are at an disadvantage. Xenos can seige them before the marines even drop. If the survivor loses, the xenos get many supplies. But if the survivor makes it out of there, they get nothing but life and a CHANCE to exterminate the hive. Sounds just like the old survivor but easier. You should make it have an incentive to hold out/go survivor.

TLDR: Please make it so that if the survivors hold out till the marines drop or when the marines get to them, they have like some valuable intel on a computer or something good for REQ points.

**Comment by Swagile** [comment]
> 
> 
> Wait, whats the purpose/incentive for going survivor? Xenos have all the benefit from what I read. They can choose to attack. If the survivors leave the bunker, they are at an disadvantage. Xenos can seige them before the marines even drop. If the survivor loses, the xenos get many supplies. But if the survivor makes it out of there, they get nothing but life and a CHANCE to exterminate the hive. Sounds just like the old survivor but easier. You should make it have an incentive to hold out/go survivor.
> 
> TLDR: Please make it so that if the survivors hold out till the marines drop or when the marines get to them, they have like some valuable intel on a computer or something good for REQ points.

That was the plan; I was going to make a unbaggable intel item that survivors can drag with them that would give 200 points if put down ASRS pad, or a intel computer that gives more than other computers.

However, considering my complete lack of enthusiasm to contribute or play TGMC due to current state of events, im closing this indeterminately. Someone else can pick this up and complete it (as its mostly done), or they can do something different. I might come back in a few months or not at all, but currently, I just don't want to do anything TGMC related right now.

Thanks for the feedback, everyone.
